lib/mix/lib/mix/tasks/cmd.ex

```diff
@@ -12,13 +12,32 @@ defmodule Mix.Tasks.Cmd do
 
       mix cmd echo pwd
 
+  You can limit which apps the cmd runs in by passing the app names
+  before the cmd:
+
+      mix cmd --app1 echo pwd
+
   Aborts when a command exits with a non-zero status.
   """
+
   @spec run(OptionParser.argv) :: :ok
   def run(args) do
-    case Mix.shell.cmd(Enum.join(args, " ")) do
-      0 -> :ok
-      s -> exit(s)
+    {args, apps} = parse_apps(args)
```

Comment:
Please pass an empty list to `parse_apps` as second argument so we don't need to use the default arguments.